### ISSUE TITLE
Tabs accessibility audit

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#1236]
# 🐛 Bug Report 

Container element with `role="tablist"` is not interactive and should not be focusable.

## 🤔 Expected Behavior

1. Container element with `role="tablist"` should not have a `tabIndex` attribute.
2. Container element with `role="tablist"` should not have the `aria-disabled` attribute.
3. `role="tab"` elements that are not disabled should not include `aria-disabled="false"`.
4. Selected and not disabled `role="tab"` element should have `tabIndex={0}` by default when the tablist initializes.
5. Add aria-label to storybook examples, so that screen readers do not announce the full tablist content as context when focus enters the tablist.
6. Container element with `role="tablist"` should have `aria-orientation="vertical"` when the orientation is vertical.

## 😯 Current Behavior

1. Container element with `role="tablist"`, seems to use `useFocusWithin` to update a `tabIndex` attribute. The CollectionManager then moves focus to the appropriate Tab for the selected key: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/tabs/src/useTabs.ts#L67-L71. The `tablist` role is not interactive, and shouldn't really be focusable, even though it loses focus immediately.
2. Container element with `role="tablist"` includes the `aria-disabled` attribute, even though it is not interactive.
3. `role="tab"` elements that are not disabled include `aria-disabled="false"`.
4. Selected and not disabled `role="tab"` element initializes with `tabIndex={-1}`, that only changes on focus within.
5. In storybook examples, so that screen readers announce the full tablist content as context when focus enters the tablist.
6. Container element with `role="tablist"` does not communicate the orientation to screen reader users when the orientation is vertical, which will present confusion because the keyboard navigation behaviors differ.

## 💁 Possible Solution

PR with suggested updates: #1237

## 🔦 Context

Review Tabs for accessibility.

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |   [v3.0.0-alpha.1 ](https://reactspectrum.blob.core.windows.net/reactspectrum/879cbbdd5031a8d6ac4e91272c8e73d7dd38b0c9/storybook-17/index.html?path=/story/tabs--default)
| Browser          | Chrome Version 86.0.4240.183 (Official Build) (x86_64)
| Operating System | macOS 10.15.7 (19H2)

## 🧢 Your Company/Team

Adobe/Accessibility
